src/components/MAPI/apps/AppList/index.tsx

```diff
@@ -0,0 +1,210 @@
+  useReducer,
+  useRef,
+  useState,
+} from 'react';
+import { useSelector } from 'react-redux';
+import { getUserIsAdmin } from 'stores/main/selectors';
+import useSWR from 'swr';
+import AppsListPage from 'UI/Display/Apps/AppList/AppsListPage';
+
+import {
+  compareAppCatalogIndexAppsFns,
+  filterAppCatalogIndexApps,
+  getAppCatalogsIndexAppList,
+  getAppCatalogsIndexAppListKey,
+  IAppCatalogIndexAppList,
+  mapAppCatalogsToFacets,
+} from './utils';
+
+const SEARCH_THROTTLE_RATE_MS = 250;
+
+type SelectedCatalogsState = Record<string, boolean>;
+
+interface ISelectedCatalogsAction {
+  type: 'selectCatalog' | 'deselectCatalog';
+  name: string;
+}
+
+const selectedCatalogsReducer: React.Reducer<
+  SelectedCatalogsState,
+  ISelectedCatalogsAction
+> = (state: SelectedCatalogsState, action: ISelectedCatalogsAction) => {
+  switch (action.type) {
+    case 'selectCatalog':
+      return Object.assign({}, state, { [action.name]: true });
+    case 'deselectCatalog': {
+      const nextSelectedCatalogs = Object.assign({}, state);
+      delete nextSelectedCatalogs[action.name];
+
+      return nextSelectedCatalogs;
+    }
+    default:
+      return state;
+  }
+};
+
+const AppList: React.FC<{}> = () => {
+  const isAdmin = useSelector(getUserIsAdmin);
+
+  const clientFactory = useHttpClientFactory();
+  const auth = useAuthProvider();
+
+  const appCatalogListClient = useRef(clientFactory());
+  const appCatalogListGetOptions: applicationv1alpha1.IGetAppCatalogListOptions = useMemo(() => {
+    // Admins can see any type of catalogs.
+    if (isAdmin) return {};
+
+    return {
+      labelSelector: {
+        matchingLabels: {
+          [applicationv1alpha1.labelCatalogVisibility]: 'public',
+          [applicationv1alpha1.labelCatalogType]: 'stable',
+        },
+      },
+    };
+  }, [isAdmin]);
+
+  const {
+    data: appCatalogList,
+    error: appCatalogListError,
+    isValidating: appCatalogListIsValidating,
+  } = useSWR<applicationv1alpha1.IAppCatalogList, GenericResponse>(
+    applicationv1alpha1.getAppCatalogListKey(appCatalogListGetOptions),
+    () =>
+      applicationv1alpha1.getAppCatalogList(
+        appCatalogListClient.current,
+        auth,
+        appCatalogListGetOptions
+      )
+  );
+  const prevAppCatalogList = usePrevious(appCatalogList);
+
+  useEffect(() => {
+    if (appCatalogListError) {
+      const errorMessage = extractErrorMessage(appCatalogListError);
+
+      new FlashMessage(
+        'There was a problem loading app catalogs.',
+        messageType.ERROR,
+        messageTTL.FOREVER,
+        errorMessage
+      );
+    }
+  }, [appCatalogListError]);
+
+  const appCatalogListIsLoading =
+    typeof appCatalogList === 'undefined' && appCatalogListIsValidating;
+
+  const [searchQuery, setSearchQuery] = useState('');
+  const debouncedSearchQuery = useDebounce(
+    searchQuery,
+    SEARCH_THROTTLE_RATE_MS
+  );
+
+  const [selectedCatalogs, dispatchSelectedCatalogs] = useReducer(
```

Comment:
Fun fact: This functionality didn't work using `useState()`, because that batched together multiple state updates (such as what happens when you select or deselect all catalogs), which resulted in the map not being updated correctly.